pvlib/solarposition.py

```diff
@@ -1378,21 +1388,23 @@ def hour_angle(times, longitude, equation_of_time):
     equation_of_time_spencer71
     equation_of_time_pvcdrom
     """
-    naive_times = times.tz_localize(None)  # naive but still localized
     # hours - timezone = (times - normalized_times) - (naive_times - times)
-    hrs_minus_tzs = 1 / NS_PER_HR * (
-        2 * times.view(np.int64) - times.normalize().view(np.int64) -
-        naive_times.view(np.int64))
+    if times.tz is None:
+        times = times.tz_localize('utc')
```

Comment:
Should we `warn` about this localization? Maybe just a note. The function promises hour angle in the time zone for the longitude.

Reply:
In case it's not obvious, this localization retains the existing behavior of assuming naive inputs to be in UTC; the rewrite for the `unit` issue just happens to make it more explicit.  See for v0.9.5:

```python
In [1]: import pvlib  # v0.9.5
   ...: import pandas as pd
   ...: for tz in [None, 'utc', 'Etc/GMT+5']:
   ...:     times = pd.date_range('2019-06-01 12:00', freq='h', periods=2, tz=tz)
   ...:     eot = pvlib.solarposition.equation_of_time_spencer71(times.dayofyear)
   ...:     ha = pvlib.solarposition.hour_angle(times, -80, eot)
   ...:     print(f'hour angles, {tz=}:', ha)
   ...:
hour angles, tz=None: [-79.36192279 -64.36192279]
hour angles, tz='utc': [-79.36192279 -64.36192279]
hour angles, tz='Etc/GMT+5': [-4.36192279 10.63807721]
```

The docstring says that `times` must be localized (`Corresponding timestamps, must be localized to the timezone for the longitude.`), so I guess naive times are technically not supported.  Other solar position functions assume UTC for naive inputs, and don't warn, so I think I'm inclined to keep that behavior here as well.